### PR TITLE
Updated doc and clean up

### PR DESCRIPTION
**Pull request by abruzzi**
- fixed unit test
- updated document

### CODE REVIEW

File: template/src/__tests__/app.tsx [hunk 1]
```diff
@@ -1,8 +1,16 @@
 import { render } from "@testing-library/react";
 import App from "../app";
 
+import { createMemoryHistory } from 'history';
+import { Router } from 'react-router-dom';
+
 test("sign up form", async () => {
-  const { getByText } = render(<App />);
+  const history = createMemoryHistory()
```
Comment (mengX666) [inline template/src/__tests__/app.tsx:8]: nit: missing a semicolon at the end 😆

File: template/src/__tests__/app.tsx [hunk 2]
```diff
@@ -1,8 +1,18 @@
 import { render } from "@testing-library/react";
+import '@testing-library/jest-dom/extend-expect';
```
Comment (mengX666) [inline template/src/__tests__/app.tsx:2]: Should we add this to `template.json`?

Comment (abruzzi) [reply template/src/__tests__/app.tsx:2]: I think it's already in package `@testing-library`, so we are good (just verified locally with a clean new folder and worked, 😄 ).

Comment (mengX666) [reply template/src/__tests__/app.tsx:2]: sorry I didn't lookt at it carefully and thought they are different packages 😅  all good

**Review by mengX666 — APPROVED** [review]
LGTM ⭐

**Pull request merged**